% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1.0in]{geometry}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{helvet}
\renewcommand*\familydefault{\sfdefault}
\usepackage{setspace}
\doublespacing
\usepackage[left]{lineno}
\linenumbers
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\hypertarget{taxonomic-resolution-matters-for-microbiome-based-classification-of-colorectal-cancer}{%
\section{Taxonomic Resolution Matters for Microbiome-Based
Classification of Colorectal
Cancer}\label{taxonomic-resolution-matters-for-microbiome-based-classification-of-colorectal-cancer}}

\vspace{10mm}

\textbf{Other title options:}\\
Optimal Taxonomic Resolution for Microbiome-Based Classification of
Colorectal Cancer ASV level resolution is unnecessary for prediction of
SRNs from microbiome data \ldots{}

\vspace{10mm}

Courtney R. Armour, Begüm D.Topçuoğlu, Andrea Garretto, Patrick D.
Schloss \({^\dagger}\)

\vspace{20mm}

\({\dagger}\) To whom correspondence should be addressed:

\href{mailto:pschloss@umich.edu}{pschloss@umich.edu}

Department of Microbiology

University of Michigan

Ann Arbor, MI 48109

\vspace{20mm}

\textbf{observation format - max 1200 words, 2 figures, 25 ref}

\newpage

\hypertarget{abstract-max-250-words}{%
\subsection{Abstract (max 250 words)}\label{abstract-max-250-words}}

\hypertarget{importance-max-150-words}{%
\subsection{Importance (max 150 words)}\label{importance-max-150-words}}

\newpage

\hypertarget{introduction-250-words}{%
\subsection{Introduction (\textasciitilde250
words)}\label{introduction-250-words}}

Colorectal cancer is one of the most common cancers in men and women and
a leading cause of cancer related deaths in the United States \{\}.
Early detection and treatment are essential to increase survival rates,
but for a variety of reasons including the invasiveness and high cost of
screening (i.e.~colonoscopy), many people do not comply with recommended
screening guidelines \{\} prompting a need for low cost, non-invasive
detection methods. A growing body of research points to the gut
microbiome as a promising target for non-invasive screening to detect
screen relevant neoplasias (neoplasms?) (SRNs) consisting of
pre-cancerous polyps (i.e.~advanced adenomas) and cancer. Efforts to
realize the diagnostic potential of the gut microbiome in detecting SRNs
have focused on machine learning (ML) methods using abundances from
operational taxonomic unit (OTU) classifications based on the amplicon
sequencing of the 16S rRNA gene \{\}. However, whether this is the
optimal taxonomic resolution for classifying SRNs from microbiome data
is unknown. Additionally, recent work has pushed for the use of amplicon
sequence variants (ASVs) to replace OTUs for marker-gene analysis
because of the improved resolution with ASVs \{\}. However, whether the
additional resolution provided by ASVs is useful for ML classification
is unclear. Since ML classification relies on consistent differences
between groups, its possible that the resolution at the ASV level is too
individualized to accurately differentiate groups. Topçuoğlu \emph{et
al} \{mBio 2020\} recently demonstrated effective application of machine
learning (ML) to microbiome based classification problems and developed
a framework for applying ML practices in a more reproducible way
(mikropml). This analysis utilizes the reproducible framework developed
by Topçuoğlu \emph{et al} to quantify which ML method and taxonomic
level produce the best performing classifier for detecting SRNs from
microbiome data.

\hypertarget{results-700-words}{%
\subsection{Results (\textasciitilde700
words)}\label{results-700-words}}

Utilizing publicly available 16S rRNA sequence data from patients with
cancer and healthy controls, we generated abundance tables with Mothur
\{\} annotated to Phylum, Class, Order, Family, Genus, OTU and ASV
levels. Using the taxonomic abundance data, we quanitfied how accurately
samples could be classified as ``normal'' or ``SRN'' (i.e.~advanced
adenoma or cancer) using machine learning (methods). Across the five
machine learning methods tested, model performance tended to increase
with taxonomic level usually peaking around genus/OTU level before
dropping off slightly with ASVs (Supplemental Figure 1). Random forest
(RF) was consistently the top performer at most taxonomic levels.
{[}\textbf{TODO}: dig into literature on this: RF might be more
appropriate for microbiome analysis since its more suitable for zero
inflated data{]}. Within the RF model, the highest AUROCs were observed
for family (median AUROC: 0.687), genus (median AUROC: 0.686), and OTU
(median AUROC: 0.698) level data with no significant difference between
the three (Figure 1). Performance with ASVs (median AUROC: 0.676) was
significantly lower than OTU and genus (p \textless{} 0.05) but not
family level P = 0.06). These results were consistent with ASVs
generated with DADA2 \{\} (median AUROC: 0.66). These results suggest
that finer taxonomic resolution is not necessarily better for
differentiating a persons cancer status from the microbiome.

One hypothesis for the observation that model performance increases from
phylum to OTU level then drops slightly at ASV level is that at higher
taxonomic levels (e.g.~phylum) there are too few taxa and too much
overlap to reliably differentiate between cases and controls. At the
level of genus/OTU data there is enough data and variation but at the
ASV level, the data is too specific to individuals and doesn't overlap
enough. Examination of the prevalence of taxa in samples at each level
supports this idea. A majority of taxa are present in greater than 75\%
of samples at the phylum (67\% of taxa) and class (63\% of taxa)
levels). The opposite is observed at the OTU and ASV level where 60\%
and 53\% of taxa respectively are only present in less than 25\% of
samples (Supplemental Figure 2). \textbf{SUMMARY SENTANCE}

~~\textbf{TODO: what are important taxa, patterns along levels (e.g fuso
always in top )}

Of note, the ML pipeline includes a pre-processing step that occurs
prior to training and classifying the ML models (methods). As part of
this step, features are removed that will not provide useful information
to build the model. For example, strongly correlated features provide
the same information to build the model and thus can be collapsed.
Additionally, features with zero or near-zero variance will not help the
model differentiate groups and thus can be removed. Interestingly,
despite starting with 104106 features at the ASV level, only 478 (0.5\%)
remained after pre-processing. At the OTU level, 20079 of the 705
features (3.5\%) remained after preprocessing (Table 1).

~~\textbf{TODO: check why ASVs were removed in preprocessing (nzv?)}\\
\hspace*{0.333em}\hspace*{0.333em}(depending on answer can discuss why
features were removed - e.g.~only in one or a few samples)

\hypertarget{discussionconclusions-250-words}{%
\subsection{Discussion/Conclusions (\textasciitilde250
words)}\label{discussionconclusions-250-words}}

These results demonstrate that consideration of the appropriate
taxonomic resolution for utilizing the microbiome as a predictive tool.
In general, we found that finer resolution (e.g.~ASVs) does not add
additional sensitivity to model predicition and at the ASV level
actually impedes model performance due to the sparsity of shared taxa.
Additionally, utilizing genus or family level data could allow for
merging data generated from different 16S regions or sequencing
platforms.

\begin{itemize}
\tightlist
\item
  Overall, the fine resolution of ASVs is unnecessary for ML
  classification with microbiome data since the vast majority of the
  ASVs are removed during preprocessing and the model performance is
  diminshed compared to other taxonomic levels.
\item
  since family/genus level is just as good as otu could we use data from
  different 16S regions?
\end{itemize}

\hypertarget{materials-and-methods}{%
\subsection{Materials and Methods}\label{materials-and-methods}}

\textbf{\emph{Dataset.}} Raw 16S rRNA gene amplicon sequence data
isolated from human gut samples \{Baxter\} was downloaded from NCBI SRA
(accession \#). This dataset contains stool samples from 490 subjects.
Based on the available metadata, samples categorized as normal, high
risk normal, or adenoma were labeled ``normal'' for this analysis and
samples categorized as advanced adenoma or carcinoma were labeled as
``screen relevant neoplasia'' (SRN). This resulted in a total of 261
``normal'' samples and 229 ``SRN'' samples.

\textbf{\emph{Data processing.}} Sequence data was processed with Mothur
(1.44.3) using the SILVA reference database (v132) to produce count
tables for phylum, class, order, family, genus, OTU, and ASV following
the Schloss Lab MiSeq SOP described on the Mothur website \{\}. ASV
level data was also produced using DADA2 to ensure consistent results
with a different pipeline. Data was processed following the DADA2
pipeline, but setting pool=TRUE to infer ASVs from the whole dataset
rather than per sample. The resulting ASV table was subsampled for
consistency with the Mothur data.

\textbf{\emph{Machine Learning.}} Machine learning models were run with
the R package mikropml (v0.0.2) \{\} to predict the diagnosis category
(normal vs SRN) of each sample. Data was preprocessed to normalize
values (scale/center? range?), remove values with zero or near-zero
variance, and collapse collinear features using default parameters.
Initially the models were run with default hyperparameters, but were
expanded if the peak performance was at the edge of the hyperparameter
range. Each taxonomic model taxonomic level combination (e.g.~Random
Forest on genus) was run with 100 different seeds (1-100). Each seed
split the data into a training (80\%) and testing (20\%) set, and output
performance of the training and testing as area under the receiver
operating curve (AUROC).

To compare performance between taxonomic levels and models, P values
were calculated as previously described \{begum\}. Another output from
the mikropml package is the permuted feature importance which is
quantified by iteratively permuting each feature in the model and
assessing the change in model performance. Features are presumed to be
important if the performance of the model, measured by the AUROC,
decreases when that feature is permuted. Ranking of feature importance
was determined by ordering the features based on the change in AUROC
where features with a larger decrease in AUROC are ranked higher in
importance.

To quantify prevalence of the features, the number of samples with
non-zero abundance was divided by the total number of samples resulting
in values ranging from 0 to 1 where 0 indicates the feature is not found
in any samples, 0.5 indicates the feature is found in half of the
samples, an 1 indicating the feature is found in all of the samples.

All code is available at: \textbf{TODO: link to code}

\hypertarget{acknowledgements}{%
\subsection{Acknowledgements}\label{acknowledgements}}

\newpage

\hypertarget{figures}{%
\subsection{Figures}\label{figures}}

\includegraphics[width=0.75\textwidth,height=\textheight]{../exploratory/figures/rf_with_stats_sideways.png}

\textbf{Figure 1: Random Forest Model Performance with Significance.}
Boxplots of AUROC values on the test dataset for 100 seeds predicting
SRNs using a Random Forest model. Dashed line denotes AUROC of 0.5 which
is equivalent to random classification. Differences in AUROC values
between taxonomic levels are significant unless otherwise specified with
``NS'' (not significant)

\newpage

\hypertarget{tables}{%
\subsection{Tables}\label{tables}}

\begin{longtable}[]{@{}lrrc@{}}
\toprule
Taxonomic Level & Number of Features &
\makecell[c]{Number of Features \\ After Preprocessing} &
\makecell[c]{Percent of Features Kept \\ After Preprocessing} \\ \addlinespace
\midrule
\endhead
Phylum & 19 & 9 & 47.4 \% \\ \addlinespace
Class & 36 & 19 & 52.8 \% \\ \addlinespace
Order & 65 & 28 & 43.1 \% \\ \addlinespace
Family & 124 & 54 & 43.5 \% \\ \addlinespace
Genus & 316 & 115 & 36.4 \% \\ \addlinespace
OTU & 20079 & 705 & 3.5 \% \\ \addlinespace
ASV & 104106 & 478 & 0.5 \% \\ \addlinespace
\bottomrule
\end{longtable}

\textbf{Table 1: Summary of Features.} Overview of the number of
features at each taxonomic level before and after preprocessing as
described in the methods.

\newpage

\hypertarget{supplemental-figures}{%
\subsection{Supplemental Figures}\label{supplemental-figures}}

\includegraphics{../exploratory/figures/all_model_level.png}

\textbf{Supplemental Figure 1: Model Performance across Taxonomy.}
Boxplot of AUROC values on the test dataset for 100 seeds for each model
type across all taxonomic levels.

\includegraphics[width=\textwidth,height=0.5\textheight]{../exploratory/figures/bin_prevalence.png}\\
\textbf{Supplemental Figure 2: Prevalence of Taxa in Samples.}
Distribution of taxa prevalence in samples at each taxonomic level.

\end{document}
