---
title: "**OptiFit: a fast method for fitting amplicon sequences to existing OTUs**"
bibliography: references.bib
output:
  pdf_document:
    keep_tex: true
    includes:
      in_header: header.tex
csl: msystems.csl
fontsize: 12pt
geometry: margin=1.0in
---


```{r knitr_settings, eval=TRUE, echo=FALSE, cache=FALSE}
knitr::opts_chunk$set(
  tidy = TRUE,
  echo = FALSE,
  eval = TRUE,
  warning = FALSE,
  cache = FALSE
)

inline_hook <- function(x){
	print(x)
	if(is.numeric(x)){
		if(abs(x - round(x)) < .Machine$double.eps^0.5){
			paste(format(x,big.mark=',', digits=0, scientific=FALSE))
		} else {
			paste(format(x,big.mark=',', digits=2, nsmall=2, scientific=FALSE))
		}

	} else {
    	paste(x)      
	}
}
knitr::knit_hooks$set(inline=inline_hook)
```

\begin{center}
\vspace{25mm}

Kelly L. Sovacool${^1}$, Sarah L. Westcott${^2}$, M. Brodie Mumphrey${^1}$,
Gabrielle A. Dotson${^1}$, Patrick D. Schloss${^2\dagger}$

$1$ Department of Computational Medicine and Bioinformatics,
University of Michigan, Ann Arbor, MI 48109

$2$ Department of Microbiology and Immunology,
University of Michigan, Ann Arbor, MI 48109

\vspace{30mm}

$\dagger$ To whom correspondence should be addressed: pschloss@umich.edu

\end{center}


\newpage
\linenumbers

## Abstract

Assigning amplicon sequences to Operational Taxonomic Units (OTUs) is an
important step in characterizing the composition of microbial communities across
large datasets. OptiClust, a _de novo_ OTU clustering method in the mothur
program, has been shown to produce higher quality OTU assignments than other
methods and at comparable or faster speeds [@westcott_opticlust_2017;
@schloss_introducing_2009]. A notable difference between _de novo_ clustering
and database-dependent methods is that OTU assignments clustered with _de novo_
methods are not stable when new sequences are added to a dataset
[@westcott_novo_2015]. However, in some cases one may wish to incorporate new
samples into a previously clustered dataset without performing clustering again
on all sequences, such as when deploying a machine learning model where OTUs are
features [@topcuoglu_framework_2020]. To provide an efficient and robust method
to fit amplicon sequence data to existing OTUs, we developed the OptiFit
algorithm as a new component of the mothur program. To benchmark the OptiFit
algorithm against _de novo_ clustering with the OptiClust algorithm, we used
four published datasets isolated from soil [@johnston_metagenomics_2016], marine
[@henson_artificial_2016], mouse [@schloss_stabilization_2012], and human
[@baxter_microbiota-based_2016] samples. For each dataset, a subset of sequences
was randomly selected and clustered into OTUs with OptiClust, then the remaining
sequences were fit to the existing OTUs using the OptiFit algorithm. This was
repeated with subsets of varying sizes ranging from 10 to 90% of sequences in
order to evaluate the bounds of the dataset size required for OptiFit.
Separately, all sequences were clustered with OptiClust to provide a baseline of
OTU assignment quality and runtime performance. Each of these routines was
repeated 10 times with different random seeds to produce results that are robust
to random variation. OTU quality was evaluated using the Matthews Correlation
Coefficient (MCC) with a sequence similarity threshold of 97% as described
previously [@schloss_application_2016; @westcott_novo_2015]. On average, fitting
sequences into existing OTUs with OptiFit performed 10 times faster than _de
novo_ clustering with OptiClust, while the average MCC scores produced were
nearly indistinguishable across each dataset. The OptiFit results across subset
sizes ranging from 10 to 90% of sequences were also very similar, with slightly
higher MCC scores for larger subset sizes. Thus, OptiFit is an efficient way to
fit new sequences to existing OTUs yet without sacrificing the quality of OTU
assignments.

### Importance

\newpage

## Introduction

## Results

## Discussion

## Materials and Methods

## Acknowledgements

KLS received support from the NIH Training Program in Bioinformatics (T32 GM070449).

The funders had no role in study design, data collection and interpretation, 
or the decision to submit the work for publication.

## References
